tests/test_yaml_file_format.py

```diff
@@ -0,0 +1,62 @@
+#################################################################
+#   Libraries
+#################################################################
+import sys, os
+import pytest
+from pipeline_utils.lib import yaml_parser
+
+#################################################################
+#   Tests
+#################################################################
+def test_file_format():
+    """
+    """
+    res = [
+            {
+                "aliases": ["cgap-core:bam"],
+                "description": "format to represent aligned reads",
+                "extrafile_formats": ["bai"],
+                "file_format": "bam",
+                "institution": "/institutions/hms-dbmi/",
+                "project": "/projects/cgap-core/",
+                "standard_file_extension": "bam",
+                "status": "shared",
+                "valid_item_types": ["FileReference", "FileProcessed"]
+            },
+            {
+                "accession": 'GAPFIXRDPDK1',
+                "aliases": ["cgap-core:bam_bai"],
+                "description": "index for bam format",
+                "extrafile_formats": [],
+                "file_format": "bam_bai",
+                "institution": "/institutions/hms-dbmi/",
+                "project": "/projects/cgap-core/",
+                "standard_file_extension": "bam.bai",
+                "status": "shared",
+                "valid_item_types": ["FileReference", "FileProcessed"],
+                "uuid": '1936f246-22e1-45dc-bb5c-9cfd55537fe9'
+            }
+        ]
+
+    for i, d in enumerate(yaml_parser.load_yaml('tests/repo_correct/portal_objects/file_format.yaml')):
+        # creating JSON object
+        d_ = yaml_parser.YAMLFileFormat(d).to_json(
+                            institution='hms-dbmi',
+                            project='cgap-core'
+                            )
+        # check
+        assert d_ == res[i]
+
+def test_file_format_error():
+    """
+    """
+
+    for i, d in enumerate(yaml_parser.load_yaml('tests/repo_error/portal_objects/file_format.yaml')):
+        try:
+            # creating JSON object
+            d_ = yaml_parser.YAMLFileFormat(d).to_json(
+                                institution='hms-dbmi',
+                                project='cgap-core'
+                                )
+        except yaml_parser.ValidationError as e:
+            pass
```

Comment:
I think you're simply testing the schema validation with these sorts of tests, which is okay but also expected since it's a external tool. More generally, you're using integrated tests of sorts for the positive tests, which is also okay, though it'd be good to add some unit tests for methods that are more complex to ensure you're capturing all the cases. It's hard to tell that's happening with one or two integrated tests per class, though some of the classes are simple enough that the integrated test suffices.